test/models/profile_test.dart

```diff
@@ -0,0 +1,185 @@
+// Wird die Methode ProcessUrl aufgrufen, wird für den dort definierten Fall (in dem Beispiel client.from('drives').select('driver_id,seats')) die Antwort definiert
+// Die Datenbankabfrage an den Client wird so abgefangen und das gewünschte Ergebnis zurückgegeben.
+// Um herauszufinden welche URL durch die jeweilige Datenbankabfrage generiert wird, einfach den auskommentierten Print-Aufruf in der mockServer.Dart Datei aktivieren
+
+void main() {
+  MockUrlProcessor profileProcessor = MockUrlProcessor();
+
+  //setup muss in jeder Testklasse einmal aufgerufen werden
+  setUp(() async {
+    await MockServer.initialize();
+    MockServer.handleRequests(profileProcessor);
+  });
+
+  group('fullname', () {
+    test('returns empty if neither name nor surname are set', () async {
+      Profile profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, '');
+    });
+
+    test('returns surname if name is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.surname);
+    });
+
+    test('returns name if surname is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.name);
+    });
+
+    test('returns name and surname if both are set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter('name'),
+        surname: NullableParameter('surname'),
+      );
+      expect(profile.fullName, 'surname name');
+    });
+  });
+
+  group('isCurrentUser', () {
+    test('returns true if id is equal to current profile id', () async {
+      final profile = ProfileFactory().generateFake();
+      final currentProfile = ProfileFactory().generateFake(
+        id: profile.id,
+      );
+      SupabaseManager.setCurrentProfile(currentProfile);
+      expect(profile.isCurrentUser, true);
+    });
+
+    test('returns false if id is not equal to current profile id', () async {
+      final profile = ProfileFactory().generateFake();
+      final currentProfile = ProfileFactory().generateFake(id: profile.id! + 1);
+      SupabaseManager.setCurrentProfile(currentProfile);
+      expect(profile.isCurrentUser, false);
+    });
+  });
+
+  group('Profile.fromJson', () {
+    test('parses a profile from json', () async {
+      Map<String, dynamic> json = {
+        "id": 1,
+        "created_at": "2021-01-01T00:00:00.000Z",
+        "username": "username",
+        "email": "email",
+        "description": "description",
+        "birth_date": "2021-01-01T00:00:00.000Z",
+        "surname": "surname",
+        "name": "name",
+        "gender": 0,
+        "avatar_url": "avatar_url",
+      };
+      Profile profile = Profile.fromJson(json);
```

Comment:
is there a way to check if the json contains any other information?

Reply:
but we just defined the json in the line above? Do you mean the `toJson` method?

Reply:
yes i'm sorry i meant toJson wrong line.

Reply:
Okay, added a check for length.